src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -46,28 +74,73 @@ DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwr
     return _lineMetrics;
 }
 
-[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextFormat> DxFontRenderData::DefaultTextFormat() noexcept
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextFormat> DxFontRenderData::DefaultTextFormat()
 {
+    if (!_dwriteTextFormat)
+    {
+        // Create the font with the fractional pixel height size.
+        // It should have an integer pixel width by our math.
+        // Then below, apply the line spacing to the format to position the floating point pixel height characters
+        // into a cell that has an integer pixel height leaving some padding above/below as necessary to round them out.
+        std::wstring localeName = UserLocaleName();
+        THROW_IF_FAILED(_BuildTextFormat(_defaultFontInfo, localeName).As(&_dwriteTextFormat));
+        THROW_IF_FAILED(_dwriteTextFormat->SetLineSpacing(_lineSpacing.method, _lineSpacing.height, _lineSpacing.baseline));
+        THROW_IF_FAILED(_dwriteTextFormat->SetParagraphAlignment(DWRITE_PARAGRAPH_ALIGNMENT_NEAR));
+        THROW_IF_FAILED(_dwriteTextFormat->SetWordWrapping(DWRITE_WORD_WRAPPING_NO_WRAP));
+    }
     return _dwriteTextFormat;
 }
 
-[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontRenderData::DefaultFontFace() noexcept
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontRenderData::DefaultFontFace()
 {
+    if (!_dwriteFontFace)
+    {
+        std::wstring fontLocaleName = UserLocaleName();
+        // _ResolveFontFaceWithFallback overrides the last argument with the locale name of the font,
+        // but we should use the system's locale to render the text.
+        _dwriteFontFace = _defaultFontInfo.ResolveFontFaceWithFallback(_dwriteFactory.Get(), fontLocaleName);
+    }
+
     return _dwriteFontFace;
 }
 
-[[nodiscard]] Microsoft::WRL::ComPtr<IBoxDrawingEffect> DxFontRenderData::DefaultBoxDrawingEffect() noexcept
+[[nodiscard]] Microsoft::WRL::ComPtr<IBoxDrawingEffect> DxFontRenderData::DefaultBoxDrawingEffect()
 {
+    if (!_boxDrawingEffect)
+    {
+        // Calculate and cache the box effect for the base font. Scale is 1.0f because the base font is exactly the scale we want already.
+        THROW_IF_FAILED(s_CalculateBoxEffect(DefaultTextFormat().Get(), _glyphCell.width(), DefaultFontFace().Get(), 1.0f, &_boxDrawingEffect));
+    }
+
     return _boxDrawingEffect;
 }
 
-[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextFormat> DxFontRenderData::ItalicTextFormat() noexcept
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextFormat> DxFontRenderData::ItalicTextFormat()
 {
+    if (!_dwriteTextFormatItalic)
+    {
+        std::wstring localeName = UserLocaleName();
+        DxFontInfo fontInfoItalic = _defaultFontInfo;
+        fontInfoItalic.SetStyle(DWRITE_FONT_STYLE_ITALIC);
+        THROW_IF_FAILED(_BuildTextFormat(fontInfoItalic, localeName).As(&_dwriteTextFormatItalic));

```

Comment:
Note that the following lines are newly added to match the implementation of `DefaultTextFormat()`. I don't really know the implication of this, though. It seems fine without the following lines.

Reply:
are you suggesting I should remove them in both `DefaultTextFormat()` and `ItalicTextFormat()`, also future `SomeTextFormat`?

I mean I have a following PR that adds `BoldTextFormat` and friends. We can probably discuss this later.

Reply:
I'm not strictly suggesting that because I'm not 100% certain it's unused. It would take figuring that out first. For this PR... you don't really need to do anything for it and can just preserve the behavior to be safe.